Code/GraphMol/catch_graphmol.cpp

```diff
@@ -1486,3 +1486,118 @@ TEST_CASE("github #2890", "[bug][molops][stereo]") {
     CHECK(bond->getStereo() == Bond::STEREOANY);
     CHECK(bond->getStereoAtoms().size() == 2);
 }
+
+
+TEST_CASE("github #3150 MolOps::removeHs removes hydrides", "[bug][molops]") {
+  SmilesParserParams smilesPs;
+  smilesPs.removeHs = false;
+  
+  SECTION("Hydride ion remove Hydrides false") {
+    std::unique_ptr<RWMol> m{SmilesToMol("[H-]", smilesPs)};
+    REQUIRE(m);
+    MolOps::RemoveHsParameters ps;
+    ps.removeHydrides = false;
+    RWMol cp(*m);
+    MolOps::removeHs(cp, ps);
+    CHECK(cp.getNumAtoms() == 1);
+    CHECK(MolOps::getFormalCharge(cp) == -1);
+  }
+
+  SECTION("Hydride ion remove Hydrides true") {
+    std::unique_ptr<RWMol> m{SmilesToMol("[H-]", smilesPs)};
+    REQUIRE(m);
+    MolOps::RemoveHsParameters ps;
+    ps.removeHydrides = true;
+    RWMol cp(*m);
+    MolOps::removeHs(cp, ps);
+    CHECK(cp.getNumAtoms() == 1);
```

Comment:
Since this seems like a surprising result (there is a hydride here, but it doesn't get removed), it is worth adding a note explaining why the H wasn't removed